### PR TITLE
Single `ReviewedRestrictedApiUsage` annotation

### PR DESCRIPTION
**Pull request by fsamuel-bs**
## General
**Before this PR**:
No reason to have 2 `ReviewedRestrictedApiUsage` annotations in the codebase.

**After this PR**:

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java [hunk 1]
```diff
@@ -25,11 +25,11 @@
 import com.palantir.atlasdb.keyvalue.api.RowResult;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.spi.KeyValueServiceConfig;
-import com.palantir.atlasdb.transaction.api.annotations.ReviewedRestrictedApiUsage;
 import com.palantir.atlasdb.transaction.api.exceptions.MoreCellsPresentThanExpectedException;
 import com.palantir.atlasdb.transaction.service.TransactionService;
 import com.palantir.common.annotation.Idempotent;
 import com.palantir.common.base.BatchingVisitable;
+import com.palantir.lock.annotations.ReviewedRestrictedApiUsage;
```
Comment (jkozlowski) [inline atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:32]: Should this be in some shared/utils package? How does this annotation actually work? Does this trigger some errorprone thing?

**Review by jkozlowski — APPROVED** [review]
Feels ok, but I also don't understand what this is supposed to do

File: atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java (hunk 1 above)
Comment (fsamuel-bs) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:32]: It does, you need to add it in the method using the restricted api - see https://github.com/palantir/atlasdb/pull/7305/files#diff-3939f806d02720d19da8a6fd016bab16268d8b80b5926d39840cd799acffadc6R1891

Comment (fsamuel-bs) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:32]: > Should this be in some shared/utils package?

Think it should be in `atlasdb-commons` - `com.palantir.common.annotations` seems like a good one. But I'll not do this for now as it seems a bit more involved (would probably extract things out of `atlasdb-common` to `atlasdb-common-api` tbh to achieve that.

Comment (jkozlowski) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:32]: Where's the bit that actually ties this specific annotation to errorprone failing?

Comment (jkozlowski) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:32]: Ohhhh, I see how this works: when you use @ RestrictedApi, you specify the annotation that turns off that warning, that's cool. Alright, I think we're good

**Comment by ergo14** [comment]
Don't you want to have one for each gated method? This is just theoretical but if you have a method that uses both, the annotation will blanket allow for both

**Comment by jkozlowski** [comment]
> that uses both

Both what? I thought we're removing copies here?

**Comment by ergo14** [comment]
> > that uses both
> 
> Both what? I thought we're removing copies here?

I'm saying that we should not be removing copies. We should have different ways to allow usage for different methods, but that's not super important so I'm fine merging these

**Comment by jkozlowski** [comment]
> I'm saying that we should not be removing copies. We should have different ways to allow usage for different methods, but that's not super important so I'm fine merging these

Feels like weird duplication to me, but maybe I don't understand the concern?

**Comment by jeremyk-91** [comment]
> No reason to have 2 ReviewedRestrictedApiUsage annotations in the codebase.

I don't agree with this in the general case: suppose you have two distinct annotations W and X permitting the usage of two distinct restricted APIs Y and Z respectively. Now suppose you have a method using Y and annotated with W. If you add (whether intentionally or otherwise) a usage of Z to the same method that already uses Y, compilation will fail until you explicitly add annotation X - but if you merge W and X into say just W, then that compilation will succeed.

This one probably doesn't matter too much, though long term we should evaluate if this is what we want.

**Pull request merged**